cli/src/cluster_query.rs

```diff
@@ -820,14 +830,40 @@ pub fn process_cluster_version(rpc_client: &RpcClient, config: &CliConfig) -> Pr
     }
 }
 
-pub fn process_fees(rpc_client: &RpcClient, config: &CliConfig) -> ProcessResult {
-    let result = rpc_client.get_recent_blockhash_with_commitment(config.commitment)?;
-    let (recent_blockhash, fee_calculator, last_valid_slot) = result.value;
-    let fees = CliFees {
-        slot: result.context.slot,
-        blockhash: recent_blockhash.to_string(),
-        lamports_per_signature: fee_calculator.lamports_per_signature,
-        last_valid_slot,
+pub fn process_fees(
+    rpc_client: &RpcClient,
+    config: &CliConfig,
+    blockhash: Option<&Hash>,
+) -> ProcessResult {
+    let result = if let Some(blockhash) = blockhash {
+        let result = rpc_client
+            .get_fee_calculator_for_blockhash_with_commitment(blockhash, config.commitment)?;
+        if let Some(fee_calculator) = result.value {
+            Some(Response {
+                context: result.context,
+                value: (*blockhash, fee_calculator, 0),
```

Comment:
nits: `const DUMMY_SLOT: Slot = Slot::max_value();` ? `0` can conflict with the genesis slot?

Reply:
Can we just go straight to CliFees here, instead of special handling for slot == 0?
... oh, looks like this is the same thing @ryoqun is suggesting below. I think that would be clearer and safer.

Reply:
Yeah I should've cleaned this up as the output changes drew the cases further apart, to the point where that bit of code sharing is totally not worth it